### PR TITLE
tidb-server: change prometrics metric instance name to host_port

### PR DESCRIPTION
**Pull request by tiancaiamao**
Change the name to "host:port" instead of "host", in case that more than one TiDB 
instance runs on one machine.

PTAL @disksing @shenli @hanfei1991

### CODE REVIEW

File: tidb-server/main.go [hunk 1]
```diff
@@ -229,6 +230,14 @@ func prometheusPushClient(addr string, interval time.Duration) {
 	}
 }
 
+func instanceName() string {
+	hostname, err := os.Hostname()
+	if err != nil {
+		return "unknown"
+	}
+	return fmt.Sprintf("%s:%s", hostname, *port)
```
Comment (disksing) [inline tidb-server/main.go]: @andelf Is `:` suitable in the instance name?

Comment (disksing) [inline tidb-server/main.go]: I think '_' will be better.

**Comment by disksing** [comment]
LGTM.
cc @shenli

**Comment by shenli** [comment]
LGTM

**Pull request merged**